packages/shorebird_cli/lib/src/commands/patch/patch_android_command.dart

```diff
@@ -144,26 +146,33 @@ Please re-run the release command for this version or create a new release.''');
       return ExitCode.software.code;
     }
 
-    final shorebirdFlutterRevision = shorebirdEnv.flutterRevision;
-    if (release.flutterRevision != shorebirdFlutterRevision) {
-      logger
-        ..err('''
-Flutter revision mismatch.
-
+    if (release.flutterRevision != originalFlutterRevision) {
+      logger.info('''
 The release you are trying to patch was built with a different version of Flutter.
 
 Release Flutter Revision: ${release.flutterRevision}
-Current Flutter Revision: $shorebirdFlutterRevision
-''')
-        ..info(
-          '''
-Either create a new release using:
-  ${lightCyan.wrap('shorebird release android')}
-
-Or change your Flutter version and try again using:
-  ${lightCyan.wrap('shorebird flutter versions use ${release.flutterRevision}')}''',
-        );
-      return ExitCode.software.code;
+Current Flutter Revision: $originalFlutterRevision''');
+
+      var flutterVersionProgress = logger.progress(
+        'Switching to Flutter revision ${release.flutterRevision}',
+      );
+      await shorebirdFlutter.useRevision(revision: release.flutterRevision);
+      flutterVersionProgress.complete();
+
+      final buildProgress = logger.progress('Building patch');
+      try {
+        await buildAppBundle(flavor: flavor, target: target);
+        buildProgress.complete();
+      } on ProcessException catch (error) {
+        buildProgress.fail('Failed to build: ${error.message}');
+        return ExitCode.software.code;
+      }
+
+      flutterVersionProgress = logger.progress(
+        'Switching back to original Flutter revision $originalFlutterRevision',
```

Comment:
```suggestion
        'Reverting to Flutter revision $originalFlutterRevision',
```
nit feel free to ignore